src/components/cylc/Graph.vue

```diff
@@ -0,0 +1,1644 @@
+  fisheye: false, // whether to perform fisheye view after expand/collapse you can specify a function too
+  animate: false, // whether to animate on drawing changes you can specify a function too
+  ready: function () {}, // callback when expand/collapse initialized
+  stop: function () {
+    this.layoutStopped = true
+    this.layoutReady = false
+    this.loading = false
+    this.loading = false
+    this.setHtmlLabel(this.cy)
+    if (tippy) {
+      tippy.hide()
+    }
+  },
+  undoable: true, // and if undoRedoExtension exists,
+  cueEnabled: true, // Whether cues are enabled
+  // expandCollapseCuePosition: 'top-left', // default cue position is top left you can specify a function per node too
+  expandCollapseCuePosition: function (ele) {
+    return ele.position()
+  },
+  expandCollapseCueSize: 20, // size of expand-collapse cue
+  expandCollapseCueLineSize: 16, // size of lines used for drawing plus-minus icons
+  expandCueImage: undefined, // image of expand icon if undefined draw regular expand cue
+  collapseCueImage: undefined, // image of collapse icon if undefined draw regular collapse cue
+  expandCollapseCueSensitivity: 1 // sensitivity of expand-collapse cues
+}
+
+export default {
+  name: 'Graph',
+  props: ['workflowid'],
+  data: function () {
+    return {
+      isConnected: false,
+      config,
+      elements,
+      graphData: {
+        nodes: [],
+        edges: []
+      },
+      i: 1,
+      // vue-spinner
+      color: '#5e9aff',
+      height: '35px',
+      width: '4px',
+      margin: '2px',
+      radius: '2px',
+      size: '.2em',
+      loading: true,
+      // layout variables
+      status: 'pending',
+      layoutName: 'dagre',
+      layoutStopped: true,
+      layoutReady: false,
+      freeze: false,
+      // elements
+      nodesEdges: [],
+      workflows: [],
+      edges: [],
+      subscriptions: {},
+      workflowId: '',
+      workflowOwner: '',
+      workflowName: '',
+      suiteName: ''
+    }
+  },
+  watch: {
+    graphData: debounce(function (newVal) {
+      this.debouncer.call()
+    }, 100),
+
+    layoutName: {
+      handler (value) {
+        this.updateLayout(value)
+      }
+    },
+
+    workflows: {
+      handler: function (newval, oldval) {
+        initialised ? this.workflowUpdated(newval) : console.debug('initialising')
+        initialised = true
+      },
+      deep: true
+    }
+  },
+
+  mixins: [mixin],
+
+  metaInfo () {
+    return {
+      title: this.getPageTitle('App.graph', { name: this.workflowId })
+    }
+  },
+
+  beforeRouteLeave (to, from, next) {
+    if (tippy) {
+      tippy.hide()
+    }
+    next()
+  },
+
+  beforeDestroy () {
+    if (tippy) {
+      tippy.hide()
+    }
+    cytoscapeService.unregister(this)
+  },
+
+  mounted () {
+    console.debug(`MOUNTED called, status: ${this.status}`)
+    this.handleMounted()
+    this.$store.watch((store) => {
+      this.workflows = store.workflows
+    })
```

Comment:
This is not very conventional. Any need to use `.watch` and be called whenever the store changes (which may or may not be for the `.workflows`) and then set the local value? Instead, we could use something like Vuex's [`mapState`](https://vuex.vuejs.org/guide/state.html#the-mapstate-helper) in a computed value?